### DISCUSSION TITLE
Consolidating lerp methods

### DISCUSSION THREAD

**Discussion by pollend** [o3de/o3de#8602] [Q&A]
there is currently 3 separate math libraries in the engine. AZMath, crylegacy and MCore. I would like to start with the various implementation of lerp with mcore. I moved the lerp into math-utility but doesn't seem like a good strategy to move the rest of the functionality into a single file. what would make sense in terms organizing this a bit? There is a larger set of changes to consolidate the rest of MCore but I would like to start here. 

https://github.com/o3de/o3de/pull/7809

**Comment by hultonha** [comment]
The best person to talk to about MCore would be @amzn-jillich to try and coordinate some of this work (who should be able to take a look at this next week).

It would be even better in my mind to focus on removing the Cry math code (but it's worth keeping in mind a lot of the code that has Cry math types in it can simply be removed (red-coded) instead of ported for the most part. It's worth checking about various parts before undertaking any large changes to make sure).

In the long-term I do think it would be great to consolidate our math libraries and default to AzMath. It might be worth bringing this up in a sig-core meeting (next meeting agenda is here - feel free to add an item https://github.com/o3de/sig-core/issues/30). Other interested parties would likely be @greerdv, @mbalfour-amzn, @kberg-amzn, @amzn-rhhong and @invertednormal to name a few 🙂.
